modules/battery_manager/php/battery_manager.class.inc

```diff
@@ -0,0 +1,294 @@
+     * @param string   $status Status to be set
+     *
+     * @return \LORIS\Http\Response response
+     */
+    function toggleStatus(\Database $db, int $id, string $status)
+    {
+        try {
+            $db->update('test_battery', array('Active' => $status), array('ID' => $id));
+            return $this->response(200, 'Availability Updated Successfully');
+        } catch (\DatabaseException $e) {
+            return $this->response(400, 'Could not update entry in the Test Battery');
+        }
+    }
+
+    /**
+     * Adds a new test to the test_battery in the database
+     *
+     * @param Database $db 
+     * @param User     $user
+     * @param array    $data
+     *
+     * @return \LORIS\Http\Response response
+     */
+    function addTest(\Database $db, \User $user, array $data)
+    {
+        if (!$user->hasPermission('battery_manager_edit')) {
+            return $this->response(400, 'Edit Permission Denied');
+        }
+
+        if (!$this->isDuplicate($db, $data)) {
+            return $this->response(400, 'This Test already exists in the database');
+        }
+
+        $test = array(
+                 'Test_name'    => $data['testName'],
+                 'AgeMinDays'   => $data['ageMinDays'],
+                 'AgeMaxDays'   => $data['ageMaxDays'],
+                 'Stage'        => $data['stage'],
+                 'SubprojectID' => $data['subproject'],
+                 'Visit_label'  => $data['visitLabel'],
+                 'CenterID'     => $data['centerId'],
+                 'firstVisit'   => $data['firstVisit'],
+                 'instr_order'  => $data['instrumentOrder'],
+                 'Active'       => 'Y',
+                );
+
+        try {
+            $db->insert('test_battery', $test);
+            return $this->response(200, 'Test Submission Successful');
+        } catch (\DatabaseException $e) {
+            return $this->response(500, 'Could not add entry to the Test Battery');
+        }
+    }
+
+    /**
+     * Checks if the entry is an exact duplicate of a previous entry
+     *
+     * @param Database $db
+     * @param array $data
+     *
+     * @return boolean
+     *
+     */
+    function isDuplicate(\Database $db, array $data) {
+        // Build SQL query based on values entered by user
+        $query = "SELECT Test_name as testName,
+                         AgeMinDays as ageMinDays,
+                         AgeMaxDays as ageMaxDays,
+                         Stage as stage,
+                         SubprojectID as subproject,
+                         Visit_label as visitLabel,
+                         CenterID as centerId,
+                         firstVisit,
+                         instr_order as instrumentOrder
+                  FROM test_battery";
+        // Select duplicate entry from Test Battery
+        $entries = $db->pselect($query, array());
+
+        foreach ($entries as $entry) {
+            if ($data === $entry) {
+                return false;
```

Comment:
with the function name, I would think that `false` would be `not duplicate` and `true` would be `is duplicate`